### ISSUE TITLE
Searching on single valued attribute with multiple values brings back 0 results

### ISSUE THREAD

**Issue by gsteimer** [tsgrp/HPI#1305]
If you setup HPI search to allow the user to enter multiple fields for a single valued attribute, the search should `OR` the criteria together.  So, for example, searching on this:

![image](https://cloud.githubusercontent.com/assets/1377228/22842424/c80cd4c0-ef9a-11e6-8fbd-38a9d8634415.png)

Should produce a DQL like this:

```sql
... ( UPPER(a_status) = 'Approved' OR UPPER(a_status) = 'Effective') ...
```

Note the parens - we don't want to mess with the `AND`ing between attributes.  

The current code doesn't parse out the multiple values properly and comes out looking like this:

```sql
... UPPER(a_status) ='APPROVED~|~EFFECTIVE' ...
```

It looks like the `DctmQueryImpl` isn't properly parsing out multiple values for a single valued field.

Note - this works properly for Solr, Alfresco, and HBase.

After fixing this, we should also test multiple values against a repeating field.  This works now; the query should look something like this:

```sql
... ( ANY  UPPER(keywords)  like '%ONE%' OR  ANY  UPPER(keywords)  like '%TWO%') ...
```

Note that the `LIKE` in the query above is because I was searching on a text field (whereas the `a_status` 
values were from a picklist - hence the `=`)

**Comment by gpolonus** [comment]
DOCUMENTUM QUERY - Fixed OC query parsing in DCTMQueryImpl.java. Previously, when multiple values were given to search on a single-valued field, the DQL query would be built to match on "VALUE1~|~VALUE2" without splitting on the "~|~". Now parsing occurs for both repeating and non-repeating attributes so long as the delimiter "~|~" exists. Also did some restructuring to the existing code. Fixed with commit #15987 CR: George Stiemer

**Comment by gsteimer** [comment]
Reopening @gpolonus - looks like if you search multiple repeating attributes, it's fine.  But if I only search one, it fails.  Seems to not be adding the `any` keyword to the query:

Ex: searching on one `keyword` generates this query:

```sql
Select ... FROM aw_controlled_doc   WHERE  UPPER(keywords)  like '%THREE%' ORDER BY object_name ASC ENABLE(NOFTDQL)
```

**Comment by gpolonus** [comment]
DOCUMENTUM QUERY - Re-fixed because the logic surrounding repeating attributes was not treating single values correctly. Fixed with revision #15993 CR: George Steimer
